aten/src/ATen/native/SharedReduceOps.h

```diff
@@ -201,8 +201,8 @@ struct AbsMinOps {
     return MIN(a, b);
   }
 
-  inline C10_DEVICE acc_t project(acc_t a) const {
-    return a;
+  inline C10_DEVICE scalar_t project(acc_t a) const {
```

Comment:
I think `project` was keeping `acc_t` on purpose, to keep intermediate results that need to be written out in the `acc_t`. 
I don't want to see changes to this file, because it's clearly working for CUDA, and is producing correct results on cuda, so that means that cpu should be able to do cpu-side fixes too